### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by rbabu-ppk**

### CODE REVIEW

File: backend/functions/department.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    .on("data", async (data) => {
+      const departmentName = data.departmentName;
+      const departmentId = data.departmentId;
+      departments.push({ departmentName, departmentId });
+    })
+    .on("end", async () => {
+      try {
+        for (const department of departments) {
+          const existingDepartment = await departmentModel.findOne({
+            departmentName: department.departmentName,
```
Comment (dkuppu-ppk) [inline backend/functions/department.ts:20]: Here check from existing map or Record<string, Department> objects dont hit the database every iteration

File: backend/functions/employee.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            employeeName: employee.firstName,
+          });
+          if (!existingDepartment) {
+            await employeeModel.create(employee);
+          } else {
+            console.log("available");
+          }
+        }
+        for (const site of sites) {
+          const existingSite = await siteModel.find({
```
Comment (dkuppu-ppk) [inline backend/functions/employee.ts:47]: Fetch all the sites at a time and store in Record<string, Site> and use that to check.

File: backend/functions/department.ts (hunk 1 above)
Comment (rbabu-ppk) [reply backend/functions/department.ts]: changes done

File: backend/functions/employee.ts (hunk 2 above)
Comment (rbabu-ppk) [reply backend/functions/employee.ts]: changes done

**Pull request merged**